src/utils/openmc/material.py

```diff
@@ -0,0 +1,491 @@
+#!/usr/bin/env python
+
+import warnings
+
+import openmc
+from openmc.checkvalue import *
+from openmc.clean_xml import *
+from xml.etree import ElementTree as ET
+from collections import MappingView
+from copy import deepcopy
+import numpy as np
+
+
+# A list of all IDs for all Materials created
+MATERIAL_IDS = list()
+
+# A static variable for auto-generated Material IDs
+AUTO_MATERIAL_ID = 10000
+
+def reset_auto_material_id():
+    global AUTO_MATERIAL_ID, MATERIAL_IDS
+    AUTO_MATERIAL_ID = 10000
+    MATERIAL_IDS = list()
+
+
+# Units for density supported by OpenMC
+DENSITY_UNITS = ['g/cm3', 'g/cc', 'kg/cm3', 'at/b-cm', 'at/cm3', 'sum']
+
+# ENDF temperatures
+ENDF_TEMPS = np.array([300, 600, 700, 900, 1200, 1500])
+
+# ENDF ZAIDs
+ENDF_ZAIDS = np.array(['70c', '71c', '72c', '73c', '74c'])
+
+# Constant for density when not needed
+NO_DENSITY = 99999.
+
+
+class Material(object):
+
+    def __init__(self, material_id=None, name=''):
+
+        # Initialize class attributes
+        self._id = None
+        self._name = ''
+        self._density = None
+        self._density_units = ''
+
+        # A dictionary of Nuclides
+        # Keys         - Nuclide names
+        # Values     - tuple (nuclide, percent, percent type)
+        self._nuclides = dict()
+
+        # A dictionary of Elements
+        # Keys         - Element names
+        # Values     - tuple (element, percent, percent type)
+        self._elements = dict()
+
+        # If specified, a list of tuples of (table name, xs identifier)
+        self._sab = list()
+
+        # If true, the material will be initialized as distributed
+        self._convert_to_distrib_comps = False
+
+        # If specified, this file will be used instead of composition values
+        self._distrib_otf_file = None
+
+        # Set the Material class attributes
+        self.set_id(material_id)
+        self.set_name(name)
+
+
+    def set_id(self, material_id=None):
+
+        global MATERIAL_IDS
+
+        # If the Material already has an ID, remove it from global list
+        if not self._id is None:
+            MATERIAL_IDS.remove(self._id)
+
+        if material_id is None:
+            global AUTO_MATERIAL_ID
```

Comment:
Also, variable scope isn't affected by branching, so having this in the if-block doesn't really do anything.